### PR TITLE
kedge 0.6.0 (new formula)

### PR DESCRIPTION
**Pull request by kadel**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/kedge.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
```
Comment (woodruffw) [inline Formula/kedge.rb]: Dangling newline here.

File: Formula/kedge.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class Kedge < Formula
+  desc "Simple, Concise & Declarative Kubernetes Applications"
+  homepage "http://kedgeproject.org"
+  url "https://github.com/kedgeproject/kedge.git",
```
Comment (woodruffw) [inline Formula/kedge.rb]: Please use a tagged release, not a git tag.

Comment (ilovezfs) [reply Formula/kedge.rb]: meaning https://github.com/kedgeproject/kedge/archive/v0.3.0.tar.gz

**Comment by kadel** [comment]
Thanks for the review.
I've updated this PR based on comments.

File: Formula/kedge.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  desc "Simple, Concise & Declarative Kubernetes Applications"
+  homepage "http://kedgeproject.org"
+  url "https://github.com/kedgeproject/kedge/archive/v0.3.0.tar.gz"
+  sha256 "ddb036ff5114f3900b68793baf9c7523b13f67fa85e6d869c0f335bac42e80df"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    mkdir_p buildpath/"src/github.com/kedgeproject"
```
Comment (ilovezfs) [inline Formula/kedge.rb]: use `mkpath` please

File: Formula/kedge.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    ln_s buildpath, buildpath/"src/github.com/kedgeproject/kedge"
+    system "make", "bin"
+    bin.install "kedge"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/kedge version")
+
+    # test a simple Kedge file
+    test_file = "name: test\n" \
```
Comment (ilovezfs) [inline Formula/kedge.rb]:
```
    (testpath/"kedgefile.yml").write <<~EOS
      name: test
      containers:
      - image: test
    EOS
```

File: Formula/kedge.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+                "containers:\n"\
+                "- image: test"
+    (testpath/"kedgefile.yml").write(test_file)
+    system "#{bin}/kedge", "generate", "-f", "kedgefile.yml"
```
Comment (ilovezfs) [inline Formula/kedge.rb]: should `assert_match`, `assert_equal`, or `assert_predicate` something has actually happened here

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by kadel** [comment]
updated

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by kadel** [comment]
Hi @ilovezfs.
Is this OK? Or should I fix something?

File: Formula/kedge.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  desc "Simple, Concise & Declarative Kubernetes Applications"
+  homepage "http://kedgeproject.org"
+  url "https://github.com/kedgeproject/kedge/archive/v0.4.0.tar.gz"
+  sha256 "383a34db1ec8faf65ec8927f1e3041998c2a414a30191339c4a39385f8717475"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    mkpath buildpath/"src/github.com/kedgeproject"
```
Comment (DomT4) [inline Formula/kedge.rb]: It's a teeny tiny thing but `(buildpath/"src/github.com/kedgeproject").mkpath` is the standard style.

Comment (kadel) [reply Formula/kedge.rb]: thx, updated

**Comment by DomT4** [comment]
Will need updating to the latest version, which is now `0.5.1`.

File: Formula/kedge.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/kedgeproject").mkpath
+    ln_s buildpath, buildpath/"src/github.com/kedgeproject/kedge"
+    system "make", "bin"
+    bin.install "kedge"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/kedge version")
```
Comment (DomT4) [inline Formula/kedge.rb]: Probably okay to drop this with the stronger test below.

File: Formula/kedge.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    # test a simple Kedge file
```
Comment (DomT4) [inline Formula/kedge.rb]: Probably obvious enough to not need the code comment.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
This needs to be refreshed with 0.5.1

**Comment by kadel** [comment]
@ilovezfs  
updated to 0.5.1

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
@kadel Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**